src/lib/common/sol-mainloop-impl-glib.c

```diff
@@ -130,15 +130,20 @@ sol_mainloop_impl_idle_del(void *handle)
     return g_source_remove((uintptr_t)handle);
 }
 
-struct fd_wrap_data {
+struct sol_fd_glib {
     bool (*cb)(void *data, int fd, unsigned int active_flags);
     const void *data;
+    int fd;
```

Comment:
starting with glib 2.36 we have, maybe move to use it and rely on newer glib? Or do it in an ifdef?
```c
GLIB_AVAILABLE_IN_2_36
gpointer             g_source_add_unix_fd    (GSource        *source,
                                              gint            fd,
                                              GIOCondition    events);
GLIB_AVAILABLE_IN_2_36
void                 g_source_modify_unix_fd (GSource        *source,
                                              gpointer        tag,
                                              GIOCondition    new_events);
GLIB_AVAILABLE_IN_2_36
void                 g_source_remove_unix_fd (GSource        *source,
                                              gpointer        tag);
GLIB_AVAILABLE_IN_2_36
GIOCondition         g_source_query_unix_fd  (GSource        *source,
                                              gpointer        tag);
```